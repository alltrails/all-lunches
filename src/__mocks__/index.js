export { default as restaurantMock } from './restaurant.mock';
export { default as restaurantsMock } from './restaurants.mock';
export { default as reactRefMock } from './reactRef.mock';
