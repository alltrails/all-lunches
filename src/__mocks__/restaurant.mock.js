/* eslint-disable sort-keys  */

export default {
  coordinates: [-122.408999, 37.7902566],
  id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
  name: "Uncle Vito's Pizza",
  photoUrl:
    'https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623',
  placeUrl:
    "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
  priceLevel: 1,
  rating: 4.3,
  supportingText: 'restaurant',
  userRatingsTotal: 1427,
};
