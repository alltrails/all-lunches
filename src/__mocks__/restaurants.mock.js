import DefaultStorePNG from 'assets/store-front.png';

export default [
  {
    coordinates: [-122.408999, 37.7902566],
    id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
    name: "Uncle Vito's Pizza",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
    priceLevel: 1,
    rating: 4.3,
    supportingText: 'restaurant',
    userRatingsTotal: 1427,
  },
  {
    coordinates: [-122.408675, 37.7900735],
    id: 'ChIJg7cNIYyAhYAR8D4ml1oItTg',
    name: "Roxanne's Cafe",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg",
    priceLevel: 2,
    rating: 4.2,
    supportingText: 'restaurant',
    userRatingsTotal: 1219,
  },
  {
    coordinates: [-122.4073542, 37.7902509],
    id: 'ChIJaa9OAYyAhYARxwgUHLNEpOY',
    name: 'Bouche',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY',
    priceLevel: 3,
    rating: 4.5,
    supportingText: 'bar',
    userRatingsTotal: 465,
  },
  {
    coordinates: [-122.409088, 37.7902449],
    id: 'ChIJvXXIOoyAhYARe-ixq3WdmrY',
    name: 'Sons & Daughters',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY',
    priceLevel: 4,
    rating: 4.6,
    supportingText: 'restaurant',
    userRatingsTotal: 249,
  },
];
