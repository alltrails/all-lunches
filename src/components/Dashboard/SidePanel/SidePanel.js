import React from 'react';
import PropTypes from 'prop-types';

import { restaurantDetailsType } from 'constants/propTypes';

import RestaurantDetails from 'components/shared/RestaurantDetails';
import HeartIcon from 'components/svg/HeartIcon';
import LoadingSpinner from 'components/shared/LoadingSpinner';

import {
  FavoriteHeartWrapper,
  FavoriteHeartButton,
  CardItem,
  CardList,
  PanelWrapper,
} from './style';

const SidePanel = ({
  favoritedItemIds,
  highlightedRestaurantId,
  isUpdatingFavorites,
  onFavoritedItemChange,
  onMouseEnter,
  onMouseLeave,
  restaurants,
}) => (
  <PanelWrapper>
    <CardList>
      {restaurants.map((restaurant) => {
        const {
          id,
          name,
          photoUrl,
          placeUrl,
          priceLevel,
          rating,
          supportingText,
          userRatingsTotal,
        } = restaurant;

        const isItemFavorited = favoritedItemIds.includes(id);
        return (
          <CardItem
            key={id}
            onMouseEnter={() => onMouseEnter(id)}
            onMouseLeave={onMouseLeave}
            isSelected={id === highlightedRestaurantId}
          >
            <RestaurantDetails
              name={name}
              photoUrl={photoUrl}
              placeUrl={placeUrl}
              priceLevel={priceLevel}
              rating={rating}
              supportingText={supportingText}
              userRatingsTotal={userRatingsTotal}
            />
            <FavoriteHeartWrapper>
              <FavoriteHeartButton onClick={() => onFavoritedItemChange(id, isItemFavorited)}>
                {isUpdatingFavorites ? (
                  <LoadingSpinner />
                ) : (
                  <HeartIcon variant={isItemFavorited ? 'green' : 'gray'} />
                )}
              </FavoriteHeartButton>
            </FavoriteHeartWrapper>
          </CardItem>
        );
      })}
    </CardList>
  </PanelWrapper>
);
SidePanel.propTypes = {
  favoritedItemIds: PropTypes.arrayOf(PropTypes.string),
  isUpdatingFavorites: PropTypes.bool.isRequired,
  highlightedRestaurantId: PropTypes.string,
  onFavoritedItemChange: PropTypes.func.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  restaurants: PropTypes.arrayOf(restaurantDetailsType),
};

SidePanel.defaultProps = {
  highlightedRestaurantId: null,
  restaurants: null,
  favoritedItemIds: [],
};

export default SidePanel;
