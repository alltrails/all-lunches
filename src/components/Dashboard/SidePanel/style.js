import styled from 'styled-components';
import { colors } from 'styles';
import TransparentButton from 'styles/components/TransparentButton';

export const PanelWrapper = styled.div`
  background: ${colors.whiteBackground};
  padding: 60px 14px 0;
  overflow-y: auto;
  height: 100vh;
`;

export const CardList = styled.ul`
  list-style-type: none;
  padding: 0;
`;

export const CardItem = styled.li`
  position: relative;
  background: white;
  margin-bottom: 10px;
  border: 2px solid ${colors.lightGray};
  border-radius: 10px;
  list-style-type: none;
  padding: 15px;
  cursor: pointer;
  overflow: hidden;

  ${(p) =>
    p.isSelected &&
    `
    border: 2px solid ${colors.green};
  `}
`;

export const FavoriteHeartButton = styled(TransparentButton)`
  position: absolute;
  top: 10%;
  right: 2%;
`;
