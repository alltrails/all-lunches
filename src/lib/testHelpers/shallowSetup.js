import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
