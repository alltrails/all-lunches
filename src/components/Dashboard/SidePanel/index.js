import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { restaurantDetailsType } from 'constants/propTypes';
import debounce from 'lib/debounce';

import * as restaurantsActions from 'store/restaurants/actions';
import {
  favoritedItemIdsSelector,
  highlightedRestaurantIdSelector,
  restaurantsFilteredListSelector,
  isUpdatingFavoritesSelector,
} from 'store/restaurants/selectors';

import SidePanel from './SidePanel';

const mapStateToProps = (state) => ({
  favoritedItemIds: favoritedItemIdsSelector(state),
  highlightedRestaurantId: highlightedRestaurantIdSelector(state),
  isUpdatingFavorites: isUpdatingFavoritesSelector(state),
  restaurants: restaurantsFilteredListSelector(state),
});

const mapDispatchToProps = {
  addFavoritedRestaurant: restaurantsActions.addFavoritedRestaurant,
  setSelectedRestaurantId: restaurantsActions.setSelectedRestaurantId,
};

const SidePanelContainer = ({
  addFavoritedRestaurant,
  favoritedItemIds,
  highlightedRestaurantId,
  isUpdatingFavorites,
  restaurants,
  setSelectedRestaurantId,
}) => {
  const handleFavoritedItemChange = (itemId, isSelected) => {
    let favoritedIds = favoritedItemIds;

    if (!isSelected) favoritedIds = [...favoritedIds, itemId];
    else favoritedIds = favoritedIds.filter((optionId) => optionId !== itemId);

    addFavoritedRestaurant(favoritedIds);
  };

  const handleMouseEnter = (itemId) => {
    debounce(setSelectedRestaurantId(itemId), 250);
  };

  const handleMouseLeave = () => {
    setSelectedRestaurantId(null);
  };

  return (
    <SidePanel
      favoritedItemIds={favoritedItemIds}
      highlightedRestaurantId={highlightedRestaurantId}
      isUpdatingFavorites={isUpdatingFavorites}
      onFavoritedItemChange={handleFavoritedItemChange}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      restaurants={restaurants}
    />
  );
};

SidePanelContainer.propTypes = {
  favoritedItemIds: PropTypes.arrayOf(PropTypes.string),
  addFavoritedRestaurant: PropTypes.func.isRequired,
  highlightedRestaurantId: PropTypes.string,
  isUpdatingFavorites: PropTypes.bool.isRequired,
  restaurants: PropTypes.arrayOf(restaurantDetailsType),
  setSelectedRestaurantId: PropTypes.func.isRequired,
};

SidePanelContainer.defaultProps = {
  highlightedRestaurantId: null,
  restaurants: null,
  favoritedItemIds: [],
};

export default connect(mapStateToProps, mapDispatchToProps)(SidePanelContainer);
