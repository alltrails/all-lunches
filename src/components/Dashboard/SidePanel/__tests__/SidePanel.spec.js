/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import { restaurantsMock } from '__mocks__';

import SidePanel from '../SidePanel';

describe('SidePanel', () => {
  const defaultProps = {
    favoritedItemIds: [],
    highlightedRestaurantId: null,
    isUpdatingFavorites: false,
    onFavoriteItemChange: () => {},
    onMouseEnter: () => {},
    onMouseLeave: () => {},
    restaurants: restaurantsMock,
  };
  const setup = shallowSetup(SidePanel, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when updating favorites', () => {
    const { wrapper } = setup({ isUpdatingFavorites: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when the first item is favorited', () => {
    const { wrapper } = setup({ favoritedItemIds: [restaurantsMock[0].id] });
    expect(wrapper).toMatchSnapshot();
  });
});
