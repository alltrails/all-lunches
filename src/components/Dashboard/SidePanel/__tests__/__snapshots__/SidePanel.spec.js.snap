// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel renders the expected view 1`] = `
<style__PanelWrapper>
  <style__CardList>
    <style__CardItem
      isSelected={false}
      key="ChIJkVLoONZ_j4ARx2vJ4umERnI"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Uncle Vito's Pizza"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
        priceLevel={1}
        rating={4.3}
        supportingText="restaurant"
        userRatingsTotal={1427}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJg7cNIYyAhYAR8D4ml1oItTg"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Roxanne's Cafe"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEB-N4JLe8ZfukWW8G8Jum72Mw6dOey3xmpS8ywrT9kWjlPfrwgcjLFYrlY1NKGCMEDPTW0I4fgrbAmcVyf5ZEsE49LXdhibUiM-E4EKVyJQ2_wmj_NRZ_i999eUks-hnCQQZAeMxbqQlz9LxXSg3XZ5PWM0u_cHkHzfydi5_0CsKf0h&3u3840&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=109601"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg"
        priceLevel={2}
        rating={4.2}
        supportingText="restaurant"
        userRatingsTotal={1219}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJaa9OAYyAhYARxwgUHLNEpOY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Bouche"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uECKcrbqCoKyZnMn5XOnFD966Cp30QU9X9dOo_T7KeS3n3P0b2K2OsoxMyu8-yJbPBSanZVr1aNknFhWD9IUWfSjlHXWttBocnuB1ntZZV8ANEq5YkZnA-rqRdfN-IIzZqpiRYlQPqkEzcVb5f8F1L8sb1UbX20jMcIu_ykyiGF8rFQZ&3u3264&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=51444"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY"
        priceLevel={3}
        rating={4.5}
        supportingText="bar"
        userRatingsTotal={465}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJvXXIOoyAhYARe-ixq3WdmrY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Sons & Daughters"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEAbOQ0teYS553dCZEA4txE2wQ53voHdUlQ3-fn5cIed26LOIVIJPWSC7tN5Ie3vhuct5Dw91rpYrP5b8qdw941fIatGOuudqCaA0Ox-MlvCb6mkyVTN5gewLdZsXw5h_QskeQCcltf6-aHXYx2Oa0O-EISmb0NDVQlYqOJmgG-CCMMz&3u1488&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=91263"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY"
        priceLevel={4}
        rating={4.6}
        supportingText="restaurant"
        userRatingsTotal={249}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
  </style__CardList>
</style__PanelWrapper>
`;

exports[`SidePanel renders the expected view when the first item is favorited 1`] = `
<style__PanelWrapper>
  <style__CardList>
    <style__CardItem
      isSelected={false}
      key="ChIJkVLoONZ_j4ARx2vJ4umERnI"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Uncle Vito's Pizza"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
        priceLevel={1}
        rating={4.3}
        supportingText="restaurant"
        userRatingsTotal={1427}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="green"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJg7cNIYyAhYAR8D4ml1oItTg"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Roxanne's Cafe"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEB-N4JLe8ZfukWW8G8Jum72Mw6dOey3xmpS8ywrT9kWjlPfrwgcjLFYrlY1NKGCMEDPTW0I4fgrbAmcVyf5ZEsE49LXdhibUiM-E4EKVyJQ2_wmj_NRZ_i999eUks-hnCQQZAeMxbqQlz9LxXSg3XZ5PWM0u_cHkHzfydi5_0CsKf0h&3u3840&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=109601"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg"
        priceLevel={2}
        rating={4.2}
        supportingText="restaurant"
        userRatingsTotal={1219}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJaa9OAYyAhYARxwgUHLNEpOY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Bouche"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uECKcrbqCoKyZnMn5XOnFD966Cp30QU9X9dOo_T7KeS3n3P0b2K2OsoxMyu8-yJbPBSanZVr1aNknFhWD9IUWfSjlHXWttBocnuB1ntZZV8ANEq5YkZnA-rqRdfN-IIzZqpiRYlQPqkEzcVb5f8F1L8sb1UbX20jMcIu_ykyiGF8rFQZ&3u3264&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=51444"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY"
        priceLevel={3}
        rating={4.5}
        supportingText="bar"
        userRatingsTotal={465}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJvXXIOoyAhYARe-ixq3WdmrY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Sons & Daughters"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEAbOQ0teYS553dCZEA4txE2wQ53voHdUlQ3-fn5cIed26LOIVIJPWSC7tN5Ie3vhuct5Dw91rpYrP5b8qdw941fIatGOuudqCaA0Ox-MlvCb6mkyVTN5gewLdZsXw5h_QskeQCcltf6-aHXYx2Oa0O-EISmb0NDVQlYqOJmgG-CCMMz&3u1488&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=91263"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY"
        priceLevel={4}
        rating={4.6}
        supportingText="restaurant"
        userRatingsTotal={249}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <HeartIcon
            variant="gray"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
  </style__CardList>
</style__PanelWrapper>
`;

exports[`SidePanel renders the expected view when updating favorites 1`] = `
<style__PanelWrapper>
  <style__CardList>
    <style__CardItem
      isSelected={false}
      key="ChIJkVLoONZ_j4ARx2vJ4umERnI"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Uncle Vito's Pizza"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
        priceLevel={1}
        rating={4.3}
        supportingText="restaurant"
        userRatingsTotal={1427}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <LoadingSpinner
            color="#428813"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJg7cNIYyAhYAR8D4ml1oItTg"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Roxanne's Cafe"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEB-N4JLe8ZfukWW8G8Jum72Mw6dOey3xmpS8ywrT9kWjlPfrwgcjLFYrlY1NKGCMEDPTW0I4fgrbAmcVyf5ZEsE49LXdhibUiM-E4EKVyJQ2_wmj_NRZ_i999eUks-hnCQQZAeMxbqQlz9LxXSg3XZ5PWM0u_cHkHzfydi5_0CsKf0h&3u3840&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=109601"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg"
        priceLevel={2}
        rating={4.2}
        supportingText="restaurant"
        userRatingsTotal={1219}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <LoadingSpinner
            color="#428813"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJaa9OAYyAhYARxwgUHLNEpOY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Bouche"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uECKcrbqCoKyZnMn5XOnFD966Cp30QU9X9dOo_T7KeS3n3P0b2K2OsoxMyu8-yJbPBSanZVr1aNknFhWD9IUWfSjlHXWttBocnuB1ntZZV8ANEq5YkZnA-rqRdfN-IIzZqpiRYlQPqkEzcVb5f8F1L8sb1UbX20jMcIu_ykyiGF8rFQZ&3u3264&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=51444"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY"
        priceLevel={3}
        rating={4.5}
        supportingText="bar"
        userRatingsTotal={465}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <LoadingSpinner
            color="#428813"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
    <style__CardItem
      isSelected={false}
      key="ChIJvXXIOoyAhYARe-ixq3WdmrY"
      onMouseEnter={[Function]}
    >
      <RestaurantDetails
        name="Sons & Daughters"
        photoUrl="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEAbOQ0teYS553dCZEA4txE2wQ53voHdUlQ3-fn5cIed26LOIVIJPWSC7tN5Ie3vhuct5Dw91rpYrP5b8qdw941fIatGOuudqCaA0Ox-MlvCb6mkyVTN5gewLdZsXw5h_QskeQCcltf6-aHXYx2Oa0O-EISmb0NDVQlYqOJmgG-CCMMz&3u1488&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=91263"
        placeUrl="https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY"
        priceLevel={4}
        rating={4.6}
        supportingText="restaurant"
        userRatingsTotal={249}
      />
      <style__FavoriteHeartWrapper>
        <style__FavoriteHeartButton
          onClick={[Function]}
        >
          <LoadingSpinner
            color="#428813"
          />
        </style__FavoriteHeartButton>
      </style__FavoriteHeartWrapper>
    </style__CardItem>
  </style__CardList>
</style__PanelWrapper>
`;
